### ISSUE TITLE
Issue with draggable annotations

### ISSUE THREAD

**Issue by jessp** [skomputer/oligrapher#21]
I was just working on redoing the annotation functionality (as per issue #16) and found that in the current demo the annotation drag and drop works unpredictably. (No console errors, but I can do a screen record if its not reproducible on your end.) I'm looking into it now, but help/advice would be appreciated.

**Comment by skomputer** [comment]
I'm noticing that annotation drag and drop only works in a single
direction. If I drag an annotation down and then back up before releasing
it, it's placed at the bottom drag position. If I drag an annotation up and
then down before releasing it, it's placed at the top position. Is this the
issue you encountered, or is it something else? Either way, I'm happy to
take care of it unless you've already figured it out.

**Comment by jessp** [comment]
It would be great if you could take a look. I messed with it a bit tonight but am stuck. If it helps, it seems like maybe it has something to do with setting the values setting the `move(from, to)`. (id doesn't seem to exist--unless that's something I messed with doing other stuff...) Thanks for looking!

**Comment by jessp** [comment]
Yeah, the id thing I mentioned seems to have just been due to my alterations. I'm back to the point where things drag up but not down.

**Comment by jessp** [comment]
Final update for a bit: I was way over thinking things and overcomplicating with new code. Seems that it was mostly a matter of the target in dragover not being properly identified. I fixed the code I'm working on but it's in a fork with lots of other nonsense going on (resulting in a currently inelegant application). I can probably figure out how the fix would apply to the master branch tomorrow after work but I'm calling it a day for now.

**Comment by skomputer** [comment]
Are you able to drag an annotation straight down?

**Comment by jessp** [comment]
Yep. Rarely, the dragged item jumps down 2 spaces instead of one but otherwise it seems to work fine. I've attached a quick (but obnoxious) gif to demonstrate since I'm at other work right now. Styling is currently a bit of a mess on this fork.

![sample_drag](https://cloud.githubusercontent.com/assets/3848662/14958535/3777aa6e-1059-11e6-9a21-7885b41bfec7.gif)

**Comment by jessp** [comment]
Okay, looks like I was mistaken about the issue being present in the current master branch. Just my own muckery on a fork + the issue being present on the demo at [http://oli2.littlesis.org/](http://oli2.littlesis.org/). As long as the demo will be updated eventually, I think we're fine to close the issue and chalk up the misunderstanding to my silliness. Apologies!

**Comment by skomputer** [comment]
don't close the issue! there is a bug: if i drag a node down and then up
before letting go of it, it only moves down, and vice versa.
